runtime/src/bank.rs

```diff
@@ -4443,11 +4437,14 @@ impl Bank {
         (program_accounts_map, loaded_programs_for_txs)
     }
 
-    fn replenish_executor_cache(
+    fn replenish_executor_cache<'a>(
         &self,
-        program_owners: &[&Pubkey],
+        program_owners: &[&'a Pubkey],
         sanitized_txs: &[SanitizedTransaction],
         check_results: &mut [TransactionCheckResult],
+    ) -> (
+        HashMap<Pubkey, &'a Pubkey>,
+        HashMap<Pubkey, Arc<LoadedProgram>>,
     ) {
         let mut filter_programs_time = Measure::start("filter_programs_accounts");
         let program_accounts_map = self.rc.accounts.filter_executable_program_accounts(
```

Comment:
Should be factored out to call site, so that we don't need to return `program_accounts_map`, but that can be done in a separate PR.